atomix/storage/src/main/java/io/atomix/storage/journal/MappedJournalSegmentWriter.java

```diff
@@ -46,38 +49,41 @@
  */
 class MappedJournalSegmentWriter<E> implements JournalWriter<E> {
 
-  private final MappedByteBuffer mappedBuffer;
-  private final ByteBuffer buffer;
+  private final FileChannel channel;
+  private final MappedByteBuffer buffer;
   private final JournalSegment<E> segment;
   private final int maxEntrySize;
   private final JournalIndex index;
   private final Namespace namespace;
   private final long firstIndex;
   private Indexed<E> lastEntry;
+  private boolean isOpen = true;
 
   MappedJournalSegmentWriter(
-      final MappedByteBuffer buffer,
+      final JournalSegmentFile file,
       final JournalSegment<E> segment,
       final int maxEntrySize,
       final JournalIndex index,
       final Namespace namespace) {
-    mappedBuffer = buffer;
-    this.buffer = buffer.slice();
     this.segment = segment;
     this.maxEntrySize = maxEntrySize;
     this.index = index;
     this.namespace = namespace;
     firstIndex = segment.index();
+    channel =
+        file.openChannel(
+            StandardOpenOption.CREATE, StandardOpenOption.READ, StandardOpenOption.WRITE);
+    buffer = mapChannel(channel, segment);
```

Comment:
afaik you can close the channel once you've mapped it, no need to keep it around.

So you could do:

```java
try(final FileChannel channel = file.openChannel(
            StandardOpenOption.CREATE, StandardOpenOption.READ, StandardOpenOption.WRITE)) {
  buffer = mapChannel(channel, segment);
}
```

But verify in case I'm wrong :slightly_smiling_face: 

Or you can just use `org.agrona.IoUtil.mapExistingFile(File, MapMode, String)` and forgo opening the channel at all :slightly_smiling_face:

Reply:
I'll give this a try 👍